### PR TITLE
Traceable wrapper subclass support for deferred runtime asserts

### PR DESCRIPTION
**Pull request by jbschlosser**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #121947
* #125947
* #122836
* #125968
* #125946
* #126552
* __->__ #126198

The padded dense -> jagged conversion op has the signature:
```
_fbgemm_dense_to_jagged_forward(Tensor dense, Tensor[] offsets, SymInt? total_L=None) -> Tensor
```

when `total_L` is not specified, the meta registration has a data-dependent output shape (based on `offsets[0][-1]`). Returning an unbacked SymInt here should work in theory, but traceable wrapper subclass support is missing in later code to handle deferred runtime asserts. This PR fixes this.

### CODE REVIEW

**Review by ezyang — APPROVED** [review]
Looks good, thanks.

File: torch/fx/experimental/symbolic_shapes.py [hunk 1]
```diff
@@ -485,6 +485,18 @@ def get(self, o: Any) -> Any:
...
 
+@dataclass(frozen=True)
+class InnerTensorKey:
+    inner_name: str
+
+    def __str__(self) -> str:
+        return f".{self.inner_name}"
+
+    def get(self, o: Any) -> Any:
+        """Call the method on object"""
```
Comment (jbschlosser) [inline torch/fx/experimental/symbolic_shapes.py:496]: comment is wrong whoops

**Pull request closed without merging**